apps/api/lib/auth.js

```diff
@@ -4,22 +4,55 @@ import auth from 'express-basic-auth'
 import clownface from 'clownface'
 import $rdf from 'rdf-ext'
 import asyncMiddleware from 'middleware-async'
-import { DESCRIBE } from '@tpluscode/sparql-builder'
+import { DESCRIBE, INSERT, sparql } from '@tpluscode/sparql-builder'
 import { rdf, vcard } from '@tpluscode/rdf-ns-builders'
 import { isNamedNode } from 'is-graph-pointer'
+import onetime from 'onetime'
 
 const require = createRequire(import.meta.url)
 
 export function basic({ client }) {
+  const credentials = loadUsers()
+    .map(([name, { password }]) => ([name, password]))
+
   return Router()
     .use(auth({
-      users: loadUsers(),
+      users: Object.fromEntries(credentials),
       challenge: true,
       realm: 'view builder',
     }))
     .use(setAgent(client))
 }
 
+export function populateUsers({ client }) {
+  const insertUserData = onetime(async (req) => {
+    const users = loadUsers()
+
+    const query = users.reduce((previous, [id, { name }]) => {
+      const userId = req.rdf.namedNode(`/user/${id}`)
+
+      return sparql`${previous}
+      
+      DROP SILENT GRAPH ${userId};
+      ${INSERT.DATA`
+        GRAPH ${userId} {
+          ${userId} a ${vcard.Individual} ; ${vcard.hasUID} "${id}" ; ${vcard.name} "${name || id}" .
+        }`};`
+    }, sparql``)
+
+    await client.query.update(query.toString())
+  })
+
+  return (req, res, next) => {
+    insertUserData(req)
+      .then(next)
+      .catch((e) => {
+        req.knossos.log(e)
+        next()
+      })
+  }
+}
```

Comment:
Currently there is no "app-start" hook so this middleware is set to run at the beginning of every API request but I wrapped the query in `onetime` so that is only inserts the data once

The query is to `DROP GRAPH` and the `INSERT DATA` for every entry from `creds.json`